### PR TITLE
Add MIME Checks for link rel=manifest

### PR DESCRIPTION
**Pull request by marcoscaceres**
Part of #24389

### CODE REVIEW

File: html/links/manifest/mime-type-application-json.headers [hunk 1]
```diff
@@ -0,0 +1 @@
+Content-Type: application/json; charset=utf-8
```
Comment (christianliebel) [inline html/links/manifest/mime-type-application-json.headers:1]: Filename must be `mime-type-application-json.webmanifest.headers`

File: html/links/manifest/mime-type-application-manifest+json.headers [hunk 2]
```diff
@@ -0,0 +1 @@
+Content-Type: application/manifest+json; charset=utf-8
```
Comment (christianliebel) [inline html/links/manifest/mime-type-application-manifest+json.headers:1]: Filename must be `mime-type-application-manifest+json.webmanifest.headers`

File: html/links/manifest/mime-type-text-json.headers [hunk 3]
```diff
@@ -0,0 +1 @@
+Content-Type: text/json; charset=utf-8
```
Comment (christianliebel) [inline html/links/manifest/mime-type-text-json.headers:1]: Filename must be `mime-type-text-json.webmanifest.headers`

File: html/links/manifest/mime-type-text-json-manual.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<title>Test JSON MIME Type support (text/json)</title>
+<link rel="help" href="https://html.spec.whatwg.org/#link-type-manifest" />
+<link rel="manifest" href="mime-type-text-json.webmanifest" />
+<h1>Test JSON MIME Type support for web manifest</h1>
+<p>
+  To pass, the use agent must treat the manifest valid (name is "pass"). The
+  response's Content-Type metadata is a JSON MIME type "text/json".
```
Comment (christianliebel) [inline html/links/manifest/mime-type-text-json-manual.html:8]: Do we want to test an "invalid" media type?

**Review by christianliebel — CHANGES_REQUESTED** [review]

**Review by christianliebel — APPROVED** [review]

File: html/links/manifest/mime-type-text-json-manual.html (hunk 4 above)
Comment (marcoscaceres) [reply html/links/manifest/mime-type-text-json-manual.html:8]: Good idea... will add an actual invalid type.

**Comment by stephenmcgruer** [comment]
@marcoscaceres the checks for this PR will be blocked due to the Taskcluster Checks migration. You can either rebase/merge-from-master the PR to bring in the new Taskcluster config, or just let me know when you want it merged and I can admin-merge it for you :)

**Comment by marcoscaceres** [comment]
Thanks @stephenmcgruer! I'll rebase.

**Comment by foolip** [comment]
The lint is failing because html/links/manifest/mime-type-none.html and html/links/manifest/mime-type-invalid.html are being interpreted as (misnamed) visual tests. To avoid this, one can put them in a resources/ or support/ subdirectory.

**Comment by marcoscaceres** [comment]
Thanks for the tip, @foolip! Fixed the naming... they should have been "-manual.html" - so thankfully the linter caught those!

**Pull request merged**